### DISCUSSION TITLE
How to deploy function variables with the Appwrite CLI?

### DISCUSSION THREAD

**Discussion by s3ppo** [appwrite/appwrite#4667] [Q&A] [answered]
### 👟 Reproduction steps

when i deploy a function with "vars", they are not provided in the new "variables" section of cloud functions

### 👍 Expected behavior

vars should be respected

### 👎 Actual Behavior

vars are ignored

### 🎲 Appwrite version

Version 1.0.x

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
@s3ppo would you please try with `variables` instead like:

```json
        {
            "$id": "my-function",
            "name": "my-function",
            "runtime": "dart-2.17",
            "path": "functions/my-function",
            "entrypoint": "lib/main.dart",
            "ignore": [".packages", ".dart_tool"],
            "execute": [],
            "variables": {
                "APPWRITE_FUNCTION_ENDPOINT": "https://[HOSTNAME]/v1",
                "APPWRITE_FUNCTION_API_KEY": "[API KEY]"
            },
            "events": ["databases.*.create"],
            "schedule": "",
            "timeout": 15
        }
```

Please also make sure you're on the latest appwrite CLI (`appwrite -v` should show 1.1.1).

**Reply by s3ppo** [reply]
Version is 1.1.1
---

<img width="886" alt="Bildschirmfoto 2022-11-13 um 17 19 35" src="https://user-images.githubusercontent.com/17125438/201532396-10363cd5-815f-4eff-9e03-a829ee2ff7e4.png">
thanks i can see now a warning because of the variables

- however... there is an error cause of a too lang variable? my variable contains an api key which is not longer than 8192 chars
- in the json of appwrite console it is named vars and not variables.. this should be identical in my opinion

**Reply by stnguyen90** [reply]
@s3ppo

> there is an error cause of a too lang variable

 did you pass an object like my sample above? Would you please share the function JSON from your appwrite.json, redacting anything as needed?

> in the json of appwrite console it is named vars and not variables.. this should be identical in my opinion

I think it's different because the format is different than that vars.

**Reply by stnguyen90** [reply]
> Anyway all other keys and formats are same like in console.. why is it here different?

@s3ppo, probably because variables were used like this in the CLI before 1.x. We do plan on documenting the appwrite.json better.

**Comment by s3ppo** [comment]
Ah okay you are right.. i tried to pass it with vars format like showed in console.. 

Anyway all other keys and formats are same like in console.. why is it here different? This is an exception where you have to waste a lot of time to find it in the docs 😉

**Comment by Ruboka** [comment]
Hey, still got this problem with the new version. appwrite cli 4.2.0 and appwrite 1.4.8
https://github.com/appwrite/website/issues/387
